tests/functional/test_basic.py

```diff
@@ -195,7 +189,9 @@ def test_python_makefile(pyi_builder):
 
 
 def test_set_icon(pyi_builder, data_dir):
-    icon_dir = os.path.join(data_dir, 'icons')
+    # Note that the name test_set_icon (not just test_icon) for prevents
+    # data_dir from copying data/icon to the tmpdir.
+    icon_dir = os.path.join(data_dir.strpath, 'icons')
```

Comment:
Providing `tmpdir/data` as the only path eliminates the possibility that a test will modify files in the source checkout, and also frees you from having to know which tests are read-only and which are read-write with respect to the data files.